aptos-move/block-executor/src/executor.rs

```diff
@@ -250,27 +218,34 @@ where
             scheduler.finish_abort(idx_to_validate, incarnation)
         } else {
             scheduler.finish_validation(idx_to_validate, validation_wave);
```

Comment:
finish validation internally has the check for updating the wave. We should return true if it updated the wave, and use that for arming? If it didn't there is no new information and reason to arm.